### ISSUE TITLE
Missing sigmoid as last discriminator layer

### ISSUE THREAD

**Issue by YOCB** [thstkdgus35/EDSR-PyTorch#55]
Is the discriminator missing a sigmoid as the last layer? (see [these lines](https://github.com/thstkdgus35/EDSR-PyTorch/blob/02a3fba34ec1d8335de259ca874405f9cff077b5/code/loss/discriminator.py#L33-L37))
If I'm not mistaken, the current last layer is a linear layer, which can output any value including negative ones. Isn't this problematic in the standard GAN (Goodfellow et al. 2014) formulation?

**Comment by thstkdgus35** [comment]
Hello.

PyTorch supports fused sigmoid operation, which combines sigmoid and binary cross entropy.

The fused operation is more stable than separating each operation, and please see [here](https://pytorch.org/docs/stable/nn.html#bcewithlogitsloss) for more detail.

Thank you.

**Comment by YOCB** [comment]
Thanks for the clarification!
